File: frontend/src/lib/schemas/referenceLooks.ts
import { z } from "zod";

// Zod-схемы для /reference-looks — соответствуют app/schemas/reference_looks.py.

export const MatchedItemSchema = z.object({
  slot: z.string(),
  item_id: z.string(),
  match_quality: z.number().min(0).max(1),
  match_reasons: z.array(z.string()).default([]),
});
export type MatchedItem = z.infer<typeof MatchedItemSchema>;

export const MissingSlotSchema = z.object({
  slot: z.string(),
  requires: z.record(z.unknown()).default({}),
  shopping_hint: z.string(),
});
export type MissingSlot = z.infer<typeof MissingSlotSchema>;

export const ReferenceLookSchema = z.object({
  look_id: z.string(),
  title: z.string(),
  occasion: z.string().nullable().optional(),
  image_url: z.string().nullable().optional(),
  description: z.string().nullable().optional(),
  matched_items: z.array(MatchedItemSchema).default([]),
  missing_slots: z.array(MissingSlotSchema).default([]),
  completeness: z.number().min(0).max(1),
  slot_order: z.array(z.string()).default([]),
});
export type ReferenceLook = z.infer<typeof ReferenceLookSchema>;

export const ReferenceLooksResponseSchema = z.object({
  subtype: z.string().nullable().optional(),
  looks: z.array(ReferenceLookSchema).default([]),
});
export type ReferenceLooksResponse = z.infer<typeof ReferenceLooksResponseSchema>;

File: ai-stylist-starter/app/schemas/reference_looks.py
"""Pydantic-схемы для роута ``/reference-looks`` (Фаза 7).

Контракт: каждый лук из ``config/rules/reference_looks/<subtype>.yaml``
отдаётся вместе с собранным матчем из гардероба пользователя. UI рисует
«скелет» лука по ``slot_order`` и подсвечивает закрытые/пустые слоты.
"""

from __future__ import annotations

from pydantic import BaseModel, Field


class MatchedItemOut(BaseModel):
    slot: str
    item_id: str
    match_quality: float = Field(ge=0.0, le=1.0)
    match_reasons: list[str] = Field(default_factory=list)


class MissingSlotOut(BaseModel):
    slot: str
    requires: dict = Field(default_factory=dict)
    shopping_hint: str


class ReferenceLookOut(BaseModel):
    look_id: str
    title: str
    occasion: str | None = None
    image_url: str | None = None
    description: str | None = None
    matched_items: list[MatchedItemOut] = Field(default_factory=list)
    missing_slots: list[MissingSlotOut] = Field(default_factory=list)
    completeness: float = Field(ge=0.0, le=1.0)
    slot_order: list[str] = Field(default_factory=list)


class ReferenceLooksResponse(BaseModel):
    subtype: str | None = Field(
        None,
        description=(
            "Активный подтип пользователя (из style_profile_resolver). "
            "None, если профиль ещё не определён — тогда looks=[]."
        ),
    )
    looks: list[ReferenceLookOut] = Field(default_factory=list)
